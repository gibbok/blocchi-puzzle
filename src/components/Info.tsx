import * as React from 'react';
import styled from 'styled-components';
import 'typeface-oleo-script';
import { wood } from '../assets/images';

const InfoStyled = styled.div`
  width: 100%;
  height: 8rem;
`;

const Group = styled.div`
  font-family: 'Oleo Script', cursive;
  font-size: 4rem;
  color: #442217;
  opacity: 0.8;
  text-shadow: 0px 1px 0px rgba(255, 255, 255, 0.4), 0px -1px 0px rgba(0, 0, 0, 0.8);
  padding: 3.5rem 0 3.5rem 0;
`;

const Line = styled.div`
  display: flex;
  justify-content: center;
`;

const Divider = styled.div`
  position: relative;
  height: 1px;
  &::before {
    content: '';
    position: absolute;
    top: 0;
    left: 5%;
    right: 5%;
    width: 90%;
    height: 1px;
    background-image: linear-gradient(to right, transparent, rgb(48, 49, 51), transparent);
  }
  &::after {
    content: '';
    position: absolute;
    z-index: 1;
    top: -7px;
    left: calc(50% - 7px);
    width: 14px;
    height: 14px;
    transform: rotate(45deg);
    background-image: url(${wood});
    border: 1px solid rgb(48, 49, 51);
    box-shadow: inset 1px 1px 3px 0px rgba(0, 0, 0, 0.4);
  }
`;

export const Info = ({ score, level, lines }: { score: number; level: number; lines: number }) => (
  <InfoStyled>
    <Group>
      <Line>Score</Line>
      <Line>{score}</Line>
    </Group>
    <Divider />
    <Group>
      <Line>Level</Line>
      <Line>{level}</Line>
    </Group>
    <Divider />
    <Group>
      <Line>Lines</Line>
      <Line>{lines}</Line>
    </Group>
  </InfoStyled>
);
