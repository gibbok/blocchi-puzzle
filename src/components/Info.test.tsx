import React from 'react';
import { Info } from './Info';
import renderer from 'react-test-renderer';

describe('<Info />', () => {
  it('should render Info', () => {
    const tree = renderer.create(<Info score={100} level={1} lines={2} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
