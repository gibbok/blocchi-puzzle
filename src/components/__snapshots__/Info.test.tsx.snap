// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Info /> should render Info 1`] = `
<div
  className="sc-AxjAm jdsAYC"
>
  <div
    className="sc-AxirZ gVRNow"
  >
    <div
      className="sc-AxiKw bLKSBM"
    >
      Score
    </div>
    <div
      className="sc-AxiKw bLKSBM"
      data-test="score"
    >
      100
    </div>
  </div>
  <div
    className="sc-AxhCb jNrBnE"
  />
  <div
    className="sc-AxirZ gVRNow"
  >
    <div
      className="sc-AxiKw bLKSBM"
    >
      Level
    </div>
    <div
      className="sc-AxiKw bLKSBM"
      data-test="level"
    >
      1
    </div>
  </div>
  <div
    className="sc-AxhCb jNrBnE"
  />
  <div
    className="sc-AxirZ gVRNow"
  >
    <div
      className="sc-AxiKw bLKSBM"
    >
      Lines
    </div>
    <div
      className="sc-AxiKw bLKSBM"
      data-test="lines"
    >
      2
    </div>
  </div>
</div>
`;
